### PR TITLE
Add tests to make sure Firefox profile is working

### PR DESCRIPTION
**Pull request by oleg-andreyev**

### CODE REVIEW

File: lib/Firefox/FirefoxProfile.php [hunk 1]
```diff
@@ -146,7 +146,10 @@ public function encode()
         foreach ($this->preferences as $key => $value) {
             $content .= sprintf("user_pref(\"%s\", %s);\n", $key, $value);
         }
-        file_put_contents($temp_dir . '/user.js', $content);
+
+        if ($content) {
```
Comment (williamdes) [inline lib/Firefox/FirefoxProfile.php:150]: Could you make this a bit more specific?

**Review by williamdes — APPROVED** [review]

**Review by williamdes — APPROVED** [review]

**Comment by oleg-andreyev** [comment]
https://github.com/mozilla/geckodriver/issues/1990

**Comment by OndraM** [comment]
> [mozilla/geckodriver#1990](https://github.com/mozilla/geckodriver/issues/1990)

Hi @oleg-andreyev , sorry for the delays... The referenced Geckodriver issue seems fixed, so could the condition for the empty user.js be removed from the pull request? Also if you could please rebase the PR with current main branch, that would be great. Thanks!

**Comment by OndraM** [comment]
I refactored the tests and merged them into `FirefoxProfileTest`, as we now already have #961 . Thanks @oleg-andreyev!

**Review by OndraM — APPROVED** [review]

**Comment by oleg-andreyev** [comment]
> I refactored the tests and merged them into `FirefoxProfileTest`, as we now already have #961 . Thanks @oleg-andreyev!

Sorry, I've some how missed your previous comment. 
Thanks for handling this. <3

**Pull request merged**